src/utils.js

```diff
@@ -99,6 +104,20 @@ const readJson = async (path) => {
   return JSON.parse(await readFile(path))
 }
 
+const filterRegexFlags = (flags) => {
+  const finalFlags = new Set()
+  let mainFlags = ["g", "i", "m", "d", "s", "u", "y"]
+
+  flags.split("").forEach((item) => {
+    if (mainFlags.includes(item)) {
+      finalFlags.add(item)
+    }
+  })
```

Comment:
this way is only O(n), your way is O(n**2).
```suggestion
  const finalFlags = new Set(["g", "i", "m", "d", "s", "u", "y"])

  flags.split('').forEach((item) => {
    finalFlags.delete(item)
  })
```

Reply:
The purpose of this code is just to clean up the user input, so that it only accepts the allowed flags. that means that if a user sends "giz" it must return "gi" z is not an allowed flag.
your code definitely doesn't do that, but thank you very much for the observation I'll try to improve it

Reply:
ohhh gotcha. in that case:
```suggestion
  const finalFlags = new Set()
  const mainFlags = new Set(["g", "i", "m", "d", "s", "u", "y"])

  flags.split("").forEach((item) => {
    if (mainFlags.has(item)) {
      finalFlags.add(item)
    }
  })
```
and then move the `mainFlags` declaration up to module level.